ledger_device_sdk/src/ecc.rs

```diff
@@ -283,9 +285,234 @@ impl<const N: usize> ECPrivateKey<N, 'W'> {
     }
 }
 
+pub struct Ed25519Stream {
+    hash: Sha2_512,
+    big_r_started: bool,
+    pub big_r: [u8; 32],
+    pub signature: [u8; 64],
+}
+
+impl Default for Ed25519Stream {
+    fn default() -> Self {
+        Ed25519Stream {
+            hash: Sha2_512::default(),
+            big_r_started: false,
+            big_r: [0u8; 32],
+            signature: [0u8; 64],
+        }
+    }
+}
+
+macro_rules! check_cx_ok {
+    ($fn_call:expr) => {{
+        let err = unsafe { $fn_call };
+        if err != CX_OK {
+            return Err(err.into());
+        }
+    }};
+}
+
+impl<const N: usize> ECPrivateKey<N, 'E'> {
+    pub fn stream_sign_init(&self, ctx: &mut Ed25519Stream) -> Result<(), CxError> {
+        // Compute prefix (see https://datatracker.ietf.org/doc/html/rfc8032#section-5.1.6, step 1)
+        match ctx.big_r_started {
```

Comment:
Perhaps having two different types, one for the initialized state and the other one that can be finalized, would help here.
Should `stream_sign_init` really be part of ECPrivateKey instead of Ed25519Stream? 
```
impl Ed25519StreamInit {
    fn stream_sign_init(self, key: &ECPrivateKey<64, 'E'>) -> Ed25519Stream {}
 }
 
 impl Ed25519Stream {
     fn sign_finalize()
     fn update()
 }
 ```

Reply:
Proposal for this https://github.com/LedgerHQ/ledger-device-rust-sdk/pull/147